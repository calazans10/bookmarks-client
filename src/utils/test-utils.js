import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../reducers';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
