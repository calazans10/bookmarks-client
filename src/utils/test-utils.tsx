import React, { ReactNode } from 'react';
import { AnyAction, Action, createStore, Store } from 'redux';
import { Provider } from 'react-redux';
import { render, RenderResult } from '@testing-library/react';
import rootReducer from '../reducers';

interface RenderWithRedux<
  S = any,
  A extends Action = AnyAction,
  I extends S = any
  > {
  (
    ui: ReactNode,
    reduxOptions?: {
      store?: Store<S, A>
      initialState?: I
    }
  ): RenderResult & {
    store: Store<S, A>
  }
}

export const renderWithRedux: RenderWithRedux = (
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
