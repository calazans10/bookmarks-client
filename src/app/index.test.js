import React from 'react';
import { wait } from '@testing-library/react';
import { renderWithRedux } from 'test-utils';
import App from './index';
import history from '../history';

it('renders without crashing', async () => {
  const { container } = renderWithRedux(<App history={history} />);
  await wait();
  expect(container.firstChild).toMatchSnapshot();
});
