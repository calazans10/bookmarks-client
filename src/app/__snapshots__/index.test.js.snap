// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="sc-AykKK bwNXth"
>
  <div
    class="sc-AykKG bRLjHj"
  >
    <p
      class="sc-AykKH hKWvaO"
    >
      <span
        class="sc-AykKI bhYJXl"
      />
       
      
    </p>
    <button
      aria-label="Close alert"
      class="sc-AykKJ cSqiHN"
      type="button"
    >
      <span>
        Close
      </span>
    </button>
  </div>
  <header
    class="sc-fzXfLO kFzgQM"
  >
    <div
      class="sc-fzXfLP ilhODR"
    >
      <div
        class="sc-AykKE evVvrJ"
      >
        <a
          class="sc-AykKF bljhJT"
          href="/"
        >
          Bookmarks
        </a>
      </div>
      <nav
        class="sc-fzXfLQ eJMyZa"
      >
        <ul>
          <li
            class="sc-fzXfLR GqOcH"
          >
            <a
              href="/sign_in"
            >
              Sign In
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <main
    class="sc-fzXfLT izLond"
  >
    <section
      class="sc-fzXfLU jBGwfK"
    >
      <h1
        class="sc-fzXfLV bNPsjm"
      >
        A smarter way to save your web pages
      </h1>
      <p>
        Bookmarks makes it simple to save web pages you find useful, interesting, or inspiring in a beautiful, easy-to-use web app.
      </p>
      <p>
        Don't have an account? 
        <a
          href="/sign_up"
        >
          Sign up here
        </a>
        .
      </p>
    </section>
  </main>
</div>
`;
